sympy/core/sympify.py

```diff
@@ -567,6 +574,10 @@ def kernS(s):
     XXX This hack should not be necessary once issue 4596 has been resolved.
     """
     hit = False
+    #initialising olds and expr to s for fixing possible unbound variable warning
+    olds = s
+    expr = sympify(s)
+    kern = '_'
```

Comment:
It is reasonable to have `olds = s` here but it should also be removed from below.

It should not be necessary to have `expr = sympify(s)` though as that is handled below. Instead the strange `for i in range(2)` loop below should be rewritten more like normal code.